src-js/analytics.js

```diff
@@ -0,0 +1,56 @@
+class AnalyticsClient {
```

Comment:
pls correct me if I'm wrong but you implemented all methods as static so we call the method on the class itself right? sort of like a singleton?

is there a reason for implement it as a class at all though? it doesn't look like you're using `this` or any other class construct. could this just be a number of analytics utility functions?

Reply:
also if we want to ensure this is used as a singleton, maybe we want to make sure multiple instances cannot be created? something like this?

`class AnalyticsClient {
  constructor() {
    throw "This is bad stuff. don't create instances mon"
}`

Reply:
This is not a Singleton but the pattern here is called Utility Class/Helper class. Because it contains only static methods, it is stateless and should not be instantiated. As you said, it could also be a bunch of functions that are populated in `window` but I prefer this pattern as it makes more sense semantically IMO